cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -516,6 +516,7 @@ public void stop(ProgramId programId) throws Exception {
             break;
           case WORKFLOW:
             programScheduler.deleteSchedules(programId);
+            programScheduler.modifySchedulesTriggeredByDeletedProgram(programId);
```

Comment:
Why a special case for Workflow? So later on if we allow scheduling of other program types, one need to remember to go back to this class and add something?